cmd/argocd/commands/repo.go

```diff
@@ -158,6 +160,7 @@ Add a HTTPS repository using username/password without verifying the server's TL
 	command.Flags().BoolVar(&insecureIgnoreHostKey, "insecure-ignore-host-key", false, "disables SSH strict host key checking (deprecated, use --insecure-skip-server-validation instead)")
 	command.Flags().BoolVar(&insecureSkipServerVerification, "insecure-skip-server-verification", false, "disables server certificate and host key checks")
 	command.Flags().BoolVar(&enableLfs, "enable-lfs", false, "enable git-lfs (Large File Support) on this repository")
+	command.Flags().StringArrayVar(&fetchRefspecs, "fetch-refspec", nil, "Refspecs to be used for fetching revisions from the remote repository")
```

Comment:
This can be specified multiple times, that's why I've taken the singular form: `--fetch-refspec '+refs/heads/*:refs/remotes/origin/*' --fetch-refspec '+refs/pull/*/head:refs/remotes/origin/pull-requests/*'`

Reply:
Speaking about tests... Maybe it makes sense to add an e2e test using the CLI to add a repo with fetchRefspecs and sync an app from it.